pkg/controller/pxc/upgrade.go

```diff
@@ -95,12 +95,16 @@ func (r *ReconcilePerconaXtraDBCluster) updatePod(ctx context.Context, sfs api.S
 		return errors.Wrap(err, "failed to get initImage")
 	}
 
+	errStsWillBeDeleted := errors.New("will be deleted")
+
 	err = k8sretry.RetryOnConflict(k8sretry.DefaultRetry, func() error {
 		currentSet := sfs.StatefulSet()
-		err := r.client.Get(ctx, client.ObjectKeyFromObject(currentSet), currentSet)
-		if client.IgnoreNotFound(err) != nil {
+		if err := r.client.Get(ctx, client.ObjectKeyFromObject(currentSet), currentSet); client.IgnoreNotFound(err) != nil {
 			return errors.Wrap(err, "failed to get statefulset")
 		}
+		if currentSet.DeletionTimestamp != nil {
```

Comment:
I think we can utilize is `.isZero` here i.e. `if !currentSet.DeletionTimestamp.IsZero() {`